core/services/job/models.go

```diff
@@ -893,13 +894,9 @@ const (
 )
 
 // Validate checks the workflow spec for correctness
-func (w *WorkflowSpec) Validate(ctx context.Context) error {
-	s, err := pkgworkflows.ParseWorkflowSpecYaml(w.Workflow)
+func (w *WorkflowSpec) Validate(ctx context.Context, logger logger.Logger) error {
+	s, err := w.SDKSpec(ctx, logger)
```

Comment:
This still seems to be conflating things. We are evidently not constructing something with a logger in order to run it. We are just accessing the `Owner` and `Name` fields? Is there a different path where `s` actually does something interesting? Could we separate that logic from this case where only some minimal metadata is used?

Reply:
The workflow binary (if it's WASM) will get run both ways.  I can pass the null logger if you want here.

Reply:
Null logger might be OK, but some kind of buffered logger that the caller could choose how to handle (e.g. print as detailed error message) seems more technically correct, if there is actually a possibility of logging :shrug: I'm not sure we have a type for that lying around though?